recipes/protobuf/all/conanfile.py

```diff
@@ -79,30 +136,79 @@ def package(self):
         if self.settings.os == "Windows":
             if self.options.shared:
-                self.cpp_info.defines = ["PROTOBUF_USE_DLLS"]
-        self.cpp_info.names["cmake_find_package"] = "Protobuf"
-        self.cpp_info.names["cmake_find_package_multi"] = "Protobuf"
-        self.cpp_info.builddirs = [
+                self.cpp_info.components["libprotobuf"].defines = ["PROTOBUF_USE_DLLS"]
+
+        self.cpp_info.components["libprotobuf"].builddirs = [
             self._cmake_install_base_path,
         ]
-        self.cpp_info.build_modules = [
+
+        self.cpp_info.components["libprotobuf"].builddirs = [self._cmake_install_base_path]
+        self.cpp_info.components["libprotobuf"].build_modules.extend([
             os.path.join(self._cmake_install_base_path, "protobuf-generate.cmake"),
             os.path.join(self._cmake_install_base_path, "protobuf-module.cmake"),
             os.path.join(self._cmake_install_base_path, "protobuf-options.cmake"),
-        ]
+        ])
+
+        self.cpp_info.components["libprotoc"].name = "libprotoc"
+        self.cpp_info.components["libprotoc"].libs = [lib_prefix + "protoc" + lib_suffix]
+        self.cpp_info.components["libprotoc"].requires = ["libprotobuf"]
+
+        self.cpp_info.components["protoc"].name = "protoc"
+        self.cpp_info.components["protoc"].requires.extend(["libprotoc", "libprotobuf"])
+
         bindir = os.path.join(self.package_folder, "bin")
         self.output.info("Appending PATH environment variable: {}".format(bindir))
         self.env_info.PATH.append(bindir)
 
+        if self.options.lite:
+            self.cpp_info.components["libprotobuf-lite"].names["cmake_find_package"] = "libprotobuf-lite"
+            self.cpp_info.components["libprotobuf-lite"].names["cmake_find_package_multi"] = "libprotobuf-lite"
+            self.cpp_info.components["libprotobuf-lite"].names["pkg_config"] = "protobuf-lite"
+            self.cpp_info.components["libprotobuf-lite"].libs = [lib_prefix + "protobuf-lite" + lib_suffix]
+            if self.settings.os == "Linux":
+                self.cpp_info.components["libprotobuf-lite"].system_libs.append("pthread")
+                if self._is_clang_x86 or "arm" in str(self.settings.arch):
+                    self.cpp_info.components["libprotobuf-lite"].system_libs.append("atomic")
+            if self.settings.os == "Windows":
+                if self.options.shared:
+                    self.cpp_info.components["libprotobuf-lite"].defines = ["PROTOBUF_USE_DLLS"]
+
+            self.cpp_info.components["libprotobuf-lite"].builddirs = [self._cmake_install_base_path]
+            self.cpp_info.components["libprotobuf-lite"].build_modules.extend([
+                os.path.join(self._cmake_install_base_path, "protobuf-generate.cmake"),
+                os.path.join(self._cmake_install_base_path, "protobuf-module.cmake"),
+                os.path.join(self._cmake_install_base_path, "protobuf-options.cmake"),
+            ])
```

Comment:
Is it correct to only include these files when `lite=True`?
Doesn't the hooks complain when `lite=False`, because it will find `.cmake` files, not added as a build module?

Reply:
It is also included for the `libprotobuf` module

Reply:
Ah thanks, I see.
Now I'm curious whether these modulse will be included once or twice.